src/cryptography/hazmat/bindings/openssl/binding.py

```diff
@@ -4,10 +4,69 @@
 
 from __future__ import absolute_import, division, print_function
 
+import os
 import threading
 
 from cryptography.hazmat.bindings._openssl import ffi, lib
 
+_osrandom_engine_id = ffi.new("const char[]", b"osrandom")
+_osrandom_engine_name = ffi.new("const char[]", b"osrandom_engine")
+
+@ffi.callback("int (*)(unsigned char *, int)", error=0)
+def osrandom_rand_bytes(buf, size):
+    signed = ffi.cast("char*", buf)
+    result = os.urandom(size)
+    signed[0:size] = result
+    return 1
+
+
+@ffi.callback("int (*)(unsigned char *, int)", error=0)
+def osrandom_pseudo_rand_bytes(buf, size):
+    result = osrandom_rand_bytes(buf, size)
+    if result == 0:
+        return -1
+    else:
+        return result
+
+
+@ffi.callback("int (*)(void)")
+def osrandom_rand_status():
+    return 1
+
+
+method = ffi.new(
+    "RAND_METHOD*", dict(bytes=osrandom_rand_bytes,
+                         pseudorand=osrandom_pseudo_rand_bytes,
+                         status=osrandom_rand_status)
+)
+
+
+def _register_osrandom_engine():
+    assert lib.ERR_peek_error() == 0
+    looked_up_engine = lib.ENGINE_by_id(_osrandom_engine_id)
+    if looked_up_engine != ffi.NULL:
+        assert lib.ERR_peek_error() == 0
+        return 2
+
+    lib.ERR_clear_error()
+
+    engine = lib.ENGINE_new()
+    try:
+        result = lib.ENGINE_set_id(engine, _osrandom_engine_id)
+        assert result == 1
+        result = lib.ENGINE_set_name(engine, _osrandom_engine_name)
+        assert result == 1
+        result = lib.ENGINE_set_RAND(engine, method)
+        assert result == 1
+        result = lib.ENGINE_add(engine)
+        assert result == 1
+    finally:
+        result = lib.ENGINE_free(engine)
+        assert result == 1
+    looked_up_engine = lib.ENGINE_by_id(_osrandom_engine_id)
+    assert looked_up_engine != ffi.NULL
```

Comment:
I added this as a way to fail fast if something went wrong.  When I had a typo in `_osrandom_engine_id` at one point the failure mode was completely perplexing, and it would crash when I double-registered the same thing.  But I guess there's no real reason to keep it there, no test that fails without it, so I'll remove it.